cogs/source.py

```diff
@@ -21,59 +43,82 @@ async def on_message(self, message: discord.Message):
         if message.author == self.bot.user:
             return
 
-        content = message.content.lower()
-
-        if 'source that' in content:
-            replied_message = message.reference.resolved if message.reference else None
+        lowercase_content = message.content.lower()
+        # Checks if the key phrase is stated
+        if SourceCog.KEY_PHRASE in lowercase_content:
+            replied_message : bool = message.reference.resolved if message.reference else None
+            # Checks if the user actually replied to a message
             if replied_message:
                 relevant_file = await self.search_files(replied_message.content)
+                # Has a relevant file been found?
                 if relevant_file:
                     file_content = await self.get_file_content(relevant_file)
+                    # Can the file content be extracted?
                     if file_content:
                         channel = message.channel
                         await channel.send(f"Content of the most relevant file ({relevant_file}):\n{file_content}\n")
 
-    async def search_files(self, input_text):
-        relevant_file = None
-        max_similarity = 0.0
-        file_dir = "./BSF-bot-data/info_commands"
+    async def search_files(self, input_text : str):
+        """
+        Searches through all of the info command files to determine the most relevant file via the max similarity score. It assumes that the relevant info text file is in .txt format
+
+        Args:
+            input_text (str): Input text from the replied message
+        """
+        relevant_info_file = None
```

Comment:
This type is confusing because Python  always assumes it will be none

relevant_info_file: Path | None = None